Dockerfile

```diff
@@ -50,3 +50,9 @@ RUN update-alternatives --install /usr/bin/python python /usr/bin/python${python
     python -m pip install --upgrade pip && \
     pip install --no-cache-dir Cython numpy && \
     pip install --no-cache-dir -r requirements-test.txt
+
+ENV MKL_NUM_THREADS=2
+ENV NUMEXPR_NUM_THREADS=2
+ENV OMP_NUM_THREADS=2
+ENV OPENBLAS_NUM_THREADS=2
+ENV VECLIB_MAXIMUM_THREADS=2
```

Comment:
Without this training results are varying considerably depending on the number of cores.

Setting it to 1 would be more sensible, but for some bizarre reason doing so causes lightning_KernelSVC kernel="poly" test to fail consistently with difference in 4th decimal digit. It seams like a bug, but for now I'll leave it as 2 as this is the number of cores github actions vm has

Reply:
> Without this training results are varying considerably depending on the number of cores.

Could you please recollect in memory which libraries were affected by the different number of threads? 

Also, please add `BLIS_NUM_THREADS` env variable here.

Reply:
> but for some bizarre reason doing so causes lightning_KernelSVC kernel="poly" test to fail consistently with difference in 4th decimal digit.

I see the same thing are happening in #271. Probably it is related to numpy's random generator - they have just changed it in `1.19`.

Reply:
On my machine only test_lightgbm.py::test_regression_random_forest fails without this environment variables with
`AssertionError: 20.182372860039543 != 36.40634951405711`